File: src/integration-tests/expecteds/ts/duplicate-names.ts
export type Baz = boolean;
export type Foo = string;
/**
 *
 * array of strings is all...
 *
 */
export type UnorderedSetOfFooz1UBFn8B = Foo[];
export type Bar = number;
export type SetOfNumbers = [Bar];
export interface ObjectOfBazLEtnUJ56 {
  NotFoo?: Baz;
  [k: string]: any;
}
export type OneOfStuff = UnorderedSetOfFooz1UBFn8B | SetOfNumbers;
/**
 *
 * Generated! Represents an alias to any of the provided schemas
 *
 */
export type AnyOfFooFooObjectOfBazLEtnUJ56OneOfStuffBar = Foo | ObjectOfBazLEtnUJ56 | OneOfStuff | Bar;

File: src/integration-tests/expecteds/py/duplicate-names.py
from typing import NewType
from typing import Union
from typing import List
from typing import Tuple
from typing import TypedDict
from typing import Optional

Baz = NewType("Baz", bool)

Foo = NewType("Foo", str)
"""array of strings is all...
"""
UnorderedSetOfFooz1UBFn8B = NewType("UnorderedSetOfFooz1UBFn8B", List[Foo])

Bar = NewType("Bar", int)

SetOfNumbers = NewType("SetOfNumbers", Tuple[Bar])

class ObjectOfBazLEtnUJ56(TypedDict):
    NotFoo: Optional[Baz]

OneOfStuff = NewType("OneOfStuff", Union[UnorderedSetOfFooz1UBFn8B, SetOfNumbers])
"""Generated! Represents an alias to any of the provided schemas
"""
AnyOfFooFooObjectOfBazLEtnUJ56OneOfStuffBar = NewType("AnyOfFooFooObjectOfBazLEtnUJ56OneOfStuffBar", Union[Foo, ObjectOfBazLEtnUJ56, OneOfStuff, Bar])
